iothub/device/src/Authentication/AuthenticationWithTokenRefresh.cs

```diff
@@ -160,6 +159,8 @@ public virtual IotHubConnectionCredentials Populate(IotHubConnectionCredentials
             iotHubConnectionCredentials.SharedAccessSignature = _token;
             iotHubConnectionCredentials.SharedAccessKey = null;
             iotHubConnectionCredentials.SharedAccessKeyName = null;
+            iotHubConnectionCredentials.SasTokenTimeToLive = _suggestedTimeToLive;
+            iotHubConnectionCredentials.SasTokenRenewalBuffer = _timeBufferPercentage;
```

Comment:
Save off the user-supplied/default (as applicable) ttl and buffer values into the IotHubConnectionCredentials class.